### DISCUSSION TITLE
Is ShredOS multithreaded/ does a hyper threaded CPU change performance?

### DISCUSSION THREAD

**Discussion by lizord69** [PartialVolume/shredos.x86_64#299] [Q&A] [answered]
Hello,  I work at a MSP and we wipe all the drives from depreciated machines before they go into our recycle pile.  I recently changed the machine I had been using as my "disk killer" to a newer machine but it is technically a lower grade CPU while having higher core count. The old hardware is currently being used on another project/use and cannot be taken back :(.  I noticed worse performance (lower mb/s and longer wipe times) recently and just was curious about if the CPU truly makes much of a difference.  It's not the end of the world if this is what I'm left with.

current specs are as follows:
CPU: i5-9400 intel
MOBO: ASUS B360M-C
PSU: thermaltake SMART 600W
RAM: 16gb Samsung 2666-DDR4
Using ShredOS version 2021.08.2_23.4x86-64_0.34_20230702 on a usb 2.0 stick

Any advice on how to boost performance would be greatly appreciated!!

**Comment by PartialVolume** [comment, marked as answer]
That particular motherboard has 6 x 6Gb/sec (750MB/s) SATA ports, however it's a fairly cheap board so I'm guessing it can't achieve 6 x 750MB/s simultaneously which would be 4,500MB/s. Assuming the SATA controller can actually achieve 750MB/s divided by the six SATA ports that's a max speed of 125MB/s which is lower than probably some of your sata disks are capable of assuming the motherboard shares bandwidth between the sata ports. 

So I'm guessing due to your motherboard SATA controller you may have a I/O bottle neck that is slowing the drives down however I may be wrong.

What about the CPU?. In nwipe look at the load values. One core flat out is 1, if your processor is 4 core load would be 4 if all cores were running flat out. Also use TOP to see how much CPU is being used by Nwipe. There's not that much you can do if it's your CPU that's the bottle neck other than avoid methods with prngs. I.e just do a single zeros wipe and last pass verification. No blanking pass.

If it's the SATA controller causing the I/O bottle neck, maybe experiment by adding a  pcie sata controller card. [Sata card](https://amzn.eu/d/g1eli6B) you could then use maybe three or four SATA on the motherboard, so the individual speed of each disc goes up and three or 4 drives on the pcie sata card. However also look at your pcie spec, high end gaming boards have more pcie data lanes for faster speed. https://www.crystalrugged.com/knowledge/what-is-pcie-slots-cards-lanes/

Maybe by utilizing all 10 SATA ports you can now wipe 10 drives simultaneously so maybe that makes up for the reduced speed by wiping more drives at the same time.
